utils/loggers/__init__.py

```diff
@@ -53,95 +81,159 @@ def __init__(self, save_dir=None, weights=None, opt=None, hyp=None, logger=None,
             self.logger.info(s)
         if not clearml:
-            prefix = colorstr('ClearML: ')
+            prefix = colorstr("ClearML: ")
             s = f"{prefix}run 'pip install clearml' to automatically track, visualize and remotely train YOLOv5 🚀 in ClearML"
             self.logger.info(s)
+        if not comet_ml:
+            prefix = colorstr("Comet: ")
+            s = f"{prefix}run 'pip install comet_ml' to automatically track, visualize YOLOv5 🚀 runs in Comet"
+            self.logger.info(s)
 
         # TensorBoard
         s = self.save_dir
-        if 'tb' in self.include and not self.opt.evolve:
-            prefix = colorstr('TensorBoard: ')
-            self.logger.info(f"{prefix}Start with 'tensorboard --logdir {s.parent}', view at http://localhost:6006/")
+        if "tb" in self.include and not self.opt.evolve:
+            prefix = colorstr("TensorBoard: ")
+            self.logger.info(
+                f"{prefix}Start with 'tensorboard --logdir {s.parent}', view at http://localhost:6006/"
+            )
             self.tb = SummaryWriter(str(s))
 
         # W&B
-        if wandb and 'wandb' in self.include:
-            wandb_artifact_resume = isinstance(self.opt.resume, str) and self.opt.resume.startswith('wandb-artifact://')
-            run_id = torch.load(self.weights).get('wandb_id') if self.opt.resume and not wandb_artifact_resume else None
+        if wandb and "wandb" in self.include:
+            wandb_artifact_resume = isinstance(
+                self.opt.resume, str
+            ) and self.opt.resume.startswith("wandb-artifact://")
+            run_id = (
+                torch.load(self.weights).get("wandb_id")
+                if self.opt.resume and not wandb_artifact_resume
+                else None
+            )
             self.opt.hyp = self.hyp  # add hyperparameters
             self.wandb = WandbLogger(self.opt, run_id)
             # temp warn. because nested artifacts not supported after 0.12.10
-            if pkg.parse_version(wandb.__version__) >= pkg.parse_version('0.12.11'):
+            if pkg.parse_version(wandb.__version__) >= pkg.parse_version("0.12.11"):
                 s = "YOLOv5 temporarily requires wandb version 0.12.10 or below. Some features may not work as expected."
                 self.logger.warning(s)
         else:
             self.wandb = None
 
         # ClearML
-        if clearml and 'clearml' in self.include:
+        if clearml and "clearml" in self.include:
             self.clearml = ClearmlLogger(self.opt, self.hyp)
         else:
             self.clearml = None
 
+        # Comet
+        if comet_ml and "comet" in self.include:
+            try:
+                if isinstance(self.opt.resume, str) and self.opt.resume.startswith(
+                    "comet://"
+                ):
+                    run_id = self.opt.resume.split("/")[-1]
```

Comment:
What are the expected format of the resume? If it's a URL (like `comet://experiment-id/`), it might be cleaner to use https://docs.python.org/3/library/urllib.parse.html?highlight=urlparse#urllib.parse.urlparse